nats.go

```diff
@@ -264,7 +264,16 @@ const (
 	nuidSize = 22
 )
 
-// A Conn represents a bare connection to a nats-server.
+// A NatsPublisher publishes a NATS message.
+//
+// Publish publishes the data argument to the given subject. The data
+// argument is left untouched and needs to be correctly interpreted on
+// the receiver.
+type NATSPublisher interface {
```

Comment:
Could be just `Publisher` since it is package `nats`.
Should we add the other `Publish` functions there too?

Reply:
@kozlovic added

Reply:
I think the name would be better if it was `Publisher`, not `NATSPublisher`. Also, about adding the other ones, it was a question ;-)
Had a quick look at go-kit and don't find an equivalent interface defined, so why do you need go-nats to have this interface?
Finally, at one point there were discussion that we should have nats.Conn be an interface instead of a struct, but that would be a 2.0 type of release. This needs more internal discussion.

Reply:
if there was `Conn` as the interface it would decide my problem.
You are right, go-kit doesn't have an equivalent of `Publisher` interface. The problem is my decorator to extend `nats.Conn`. I'm trying to create nats transport like http transport and there is `interceptingWriter` implementing `http.ResponseWriter`. I wanted to create something like this.
If you are interested you can look at my fork https://github.com/kirooha/kit/blob/master/transport/nats/server.go#L195 

Maybe the better solution is creating nats transport without this decorator. I mean client code will depend on `nats.Conn` not some interface of publishing. And when there is `Conn` as the interface I will able to add decorator. What do you think?